src/state/erc20/reducer.ts

```diff
@@ -0,0 +1,24 @@
+import { buildErc20Key, Erc20State } from '.'
+import { ReducerActionType } from './actions'
+
+export function reducer(state: Erc20State, action: ReducerActionType): Erc20State {
+  switch (action.type) {
+    case 'SAVE_MULTIPLE_ERC20': {
+      // Clone current state
+      const map = new Map(state)
+
+      const { erc20s, networkId } = action.payload
+
+      erc20s.forEach((erc20) => {
```

Comment:
not sure if this logic should be in the reducer, it may be an anti-pattern and could be better in a `batchAction` where it's necessary. Thoughts @anxolin

Reply:
How would that look like?

Reply:
yes, I think is a fair concern, it also seemed like this reducer had some logic there, but I would argue this logic is a pure reducer. Ir receives the new data, and it returns the merged map with the new state

Or can u explain further your concern david?

Reply:
mm @alfetopito now I take it back, I had brushed over this and wasn't sure if this was pure or not but it is. I usually prefer `thin` reducers and `thicker` or fat actions / component logic. So in this case it would be removing the `saveMultipleErc20` action/reducer and looping + dispatching single actions inside a 
```ts
React.batchActions(() => {  
  erc20s.forEach(() => dispatch(singleErc20Action(...)))
})
```  

but i digress, that's too much rework and this is fine. just a thought tho 
@anxolin tagging u as well

Reply:
Makes sense, but what would be the advantage?
In the end, the object will be recreated on every update, doesn't matter if updating 1 or more items, no?
If we batch, there's N state updates instead of 1 (ok, they are batched, but still)

I don't mind using the batched version, just want to understand why would that be better